tst/EnergyPlus/unit/UnitarySystem.unit.cc

```diff
@@ -14750,11 +14750,11 @@ TEST_F(EnergyPlusFixture, Test_UnitarySystemModel_SubcoolReheatCoil)
                       SenOutput,
                       LatOutput);
     EXPECT_EQ(state->dataCoilCooingDX->coilCoolingDXs[0].performance.OperatingMode, 3);
-    EXPECT_NEAR(state->dataCoilCooingDX->coilCoolingDXs[0].performance.ModeRatio, 0.55356, 0.001);
+    EXPECT_NEAR(state->dataCoilCooingDX->coilCoolingDXs[0].performance.ModeRatio, 0.1991, 0.001);
     EXPECT_NEAR(thisSys->LoadSHR, 0.57154, 0.001);
-    EXPECT_NEAR(thisSys->CoilSHR, 0.4578, 0.001);
-    EXPECT_NEAR(SenOutput, -397.162, 0.1);
-    EXPECT_NEAR(LatOutput, -495.2, 0.1);
+    EXPECT_NEAR(thisSys->CoilSHR, 0.5266, 0.001);
+    EXPECT_NEAR(SenOutput, -397.032, 0.1);
+    EXPECT_NEAR(LatOutput, -338.8, 0.1);
```

Comment:
Why would the coil SHR increase? Pulling Tout back to the saturation curve at constant enthalpy should lower SHR and increase latent. Maybe the code change here is affecting a different mode while iterating to find the final operating mode = 3?

Reply:
I saw that. I can't recall what picks mode 1 or 2 right now but it feels like the choice of mode (to get to mode=3) and the iteration on solution (i.e., converging on sensible, latent or SHR) is the key to figuring out what is happening. I would agree on a follow up issue for these changes in unit tests but what bothers me here is that the outlet condition is not adjusted until after other calculations for bypass and cycling fan. I still think bottom of CalcSpeedOutput is the right place ... and would never need to be checked again.

Reply:
Looking at the method in `CalcSpeedOutput` I don't think it can ever exceed saturation. These saturation checks are trying to protect when mixing two air streams in a straight line results in a condition outside the saturation curve, like this:
![image](https://user-images.githubusercontent.com/4991075/146041169-78c07de8-26a8-4381-bf7f-f4b078567d85.png)

Reply:
So the issue here must have been cycling fan, with or without bypassing. It seems this code change should be applied to cycling fan operation.

Reply:
Sorry @mjwitte I had the concept right but the fan mode backwards. I was just thinking about this and with cycling fan the outlet condition is at full outlet and air mass flow is averaged, it's constant fan that does the averaging of inlet/outlet conditions. So this test should be moved into the if (fanOpMode == DataHVACGlobals::ContFanCycCoil) block.

Reply:
What got me thinking about this was I was trying to conceive a lab test where I could cycle the compressor and average the supply air conditions over some compressor PLR to see if I could validate this averaging method, and the Tsat issue. What is hard about this is I need a near saturated inlet condition. That is hard to get to with my labs steamer capacity, yet easy with some OA weather conditions. Also, as soon as I turn off the compressor the lab conditions go haywire (PI controlled heaters and steamer can't react fast enough) such that averaging of supply conditions would not be accurate.

Reply:
So I made the changes I suggested, and a few more subtle changes, and the unit test flipped back. Hmmm.

Reply:
Using this branch as it stands (my changes reverted), in the debugger, it looks to me as if Tout (with cycling fan mode) based on H and w:

    outletNode.Temp = Psychrometrics::PsyTdbFnHW(outletNode.Enthalpy, outletNode.HumRat);

gives a Tout that is lower than Tsat based on H and P:

    Real64 tsat = Psychrometrics::PsyTsatFnHPb(state, outletNode.Enthalpy, inletNode.Press, RoutineName);

again, Hmmm.